src/main/scala/com/github/sashjakk/http/Server.scala

```diff
@@ -0,0 +1,72 @@
+package com.github.sashjakk.http
+
+import cats.effect.{ExitCode, IO, IOApp}
+import cats.implicits._
+import com.github.sashjakk.http.Codecs._
+import com.github.sashjakk.spot._
+import com.github.sashjakk.spot.book.{BookingCreate, BookingRepo}
+import com.github.sashjakk.spot.share.{ShareCreate, SpotShareRepo}
+import com.github.sashjakk.user.{UserCreate, UserRepo, UserService}
+import fs2.Compiler.Target.forConcurrent
+import org.http4s._
+import org.http4s.circe.CirceEntityCodec.{circeEntityDecoder, circeEntityEncoder}
+import org.http4s.dsl.io._
+import org.http4s.ember.server.EmberServerBuilder
+import org.http4s.server.middleware.ErrorHandling
+
+object Server extends IOApp {
+  private def users(service: UserService[IO]): HttpRoutes[IO] = {
+    HttpRoutes.of[IO] { case request @ POST -> Root / "users" =>
+      request
+        .as[UserCreate]
+        .flatMap(service.create)
+        .redeemWith(e => BadRequest(APIError(e.getMessage)), Created(_))
+    }
+  }
+
+  private def spots(service: SpotService[IO]): HttpRoutes[IO] = {
+    HttpRoutes.of[IO] {
+      case GET -> Root / "spots" =>
+        service
+          .sharedSpots()
+          .redeemWith(e => BadRequest(APIError(e.getMessage)), Ok(_))
```

Comment:
You can use a [middleware](https://http4s.org/v0.23/docs/middleware.html) to handle this commonly.